tests/foreman/ui/test_http_proxy.py

```diff
@@ -67,3 +82,117 @@ def test_positive_create_update_delete(session, module_org, module_loc):
         # Delete http_proxy
         session.http_proxy.delete(updated_proxy_name)
         assert not session.http_proxy.search(updated_proxy_name)
+
+
+@tier2
+def test_positive_assign_http_proxy_to_products_repositories(session, module_org, module_loc):
+    """Assign HTTP Proxy to Products and Repositories.
+
+    :id: 2b803f9c-8d5d-4467-8eba-18244ebc0201
+
+    :expectedresults: HTTP Proxy is assigned to all repos present
+        in Products.
+
+    :CaseImportance: Critical
+    """
+    # create HTTP proxies
+    http_proxy_url_a = '{}:{}'.format(
+        gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999)
+    )
+    http_proxy_a = entities.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=http_proxy_url_a,
+        organization=[module_org.id],
+        location=[module_loc.id],
+    ).create()
+    http_proxy_url_b = '{}:{}'.format(
+        gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999)
+    )
+    http_proxy_b = entities.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=http_proxy_url_b,
+        organization=[module_org.id],
+        location=[module_loc.id],
+    ).create()
+    # Create products
+    product_a = entities.Product(organization=module_org.id,).create()
+    product_b = entities.Product(organization=module_org.id,).create()
+    # Create repositories from UI.
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
```

Comment:
Do you need these two lines? What would happen if you don't use them?

I thought UI uses new user that is assigned to new organization and location by default, so there is no need for you to manually select it from menu after logging in.